### PR TITLE
[PDI-17400] Vertica Bulk Loader "Abort on Error" setting not working

### PR DESCRIPTION
**Pull request by gryphendowre**
@pentaho/tatooine @pentaho-lmartins @LeonardoCoelho71950 

**NOTE:** Please read up on the ticket to understand the fix (https://jira.pentaho.com/browse/PDI-17400)

* [PDI-17400] Added Listeners for Abort On Error and Insert Direct to ROS, the metadata objects were not being updated on time and properly.
* [PDI-17400] Updated plugin code so that we handle the abort ourselves (the code previously forced us to handle it, this fix allow us to handle it appropriately). Added in a way to handle a No Abort On Error instance (drop the bad row and keep going) and to throw an exception if we did set Abort On Error.
* [PDI-17400] Updated copyrights
* [PDI-17400] Added in Exception/Rejection loggers (FileOutputStreams), opening, writing and closing.
* [PDI-17400] Added in test for StreamEncoder, verifying good and bad row data. VerticaBulkLoader to verify Abort on Error functionality. Except/Rejection logs

### CODE REVIEW

**Comment by pentaho-lmartins** [comment]
Hey @gryphendowre, lets review this together on Monday.
You need to help me understand some of this.

File: core/src/main/java/org/pentaho/di/verticabulkload/VerticaBulkLoader.java [hunk 1]
```diff
@@ -186,6 +196,66 @@ public boolean processRow( StepMetaInterface smi, StepDataInterface sdi ) throws
     return true;
   }
 
+  @VisibleForTesting
+  protected void initializeLogFiles() throws KettleException {
```
Comment (gryphendowre) [inline core/src/main/java/org/pentaho/di/verticabulkload/VerticaBulkLoader.java]: Remove protected modifiers here (and at writeExceptionRejectionLogs and closeLogFiles)

Comment (gryphendowre) [reply core/src/main/java/org/pentaho/di/verticabulkload/VerticaBulkLoader.java]: Modifiers updated

**Comment by pentaho-lmartins** [comment]
Hey @wseyler, can you take a look as well and merge if you're OK with it?

**Comment by pentaho-lmartins** [comment]
The JIRA case has a lot more detail about this fix.

**Pull request merged**